framework/src/config/mod.rs

```diff
@@ -117,55 +86,141 @@ pub struct PortConfiguration {
     pub csum: bool,
 }
 
-impl Default for PortConfiguration {
-    fn default() -> PortConfiguration {
-        PortConfiguration {
-            name: String::new(),
-            rx_queues: vec![],
-            tx_queues: vec![],
-            rxd: NUM_RXD,
-            txd: NUM_TXD,
-            loopback: false,
-            tso: false,
-            csum: false,
-        }
+impl fmt::Display for PortConfiguration {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "name: {}, rxq: {:?}, txq: {:?}, rxd: {}, txd: {}, loopback: {}, tso: {}, csum: {}",
+            self.name,
+            self.rx_queues,
+            self.tx_queues,
+            self.rxd,
+            self.txd,
+            self.loopback,
+            self.tso,
+            self.csum,
+        )
     }
 }
 
-impl PortConfiguration {
-    pub fn new_with_name(name: &str) -> PortConfiguration {
-        PortConfiguration {
-            name: String::from(name),
-            ..Default::default()
-        }
+lazy_static! {
+    pub(crate) static ref CLI_ARGS: ArgMatches<'static> = clap_app!(app =>
+        (version: "0.3.0")
+        (@arg file: -f --file +takes_value "custom configuration file")
+        (@arg name: -n --name +takes_value "DPDK process name")
+        (@group process_mode =>
+            (@arg primary: --primary conflicts_with[secondary] "run as a primary process")
+            (@arg secondary: --secondary conflicts_with[primary] "run as a secondary process")
+        )
+        (@arg primary_core: --("primary-core") +takes_value "the core to run the main thread on")
+        (@arg cores: -c --core ... +takes_value "core that NetBricks can use")
+        (@arg ports: -p --port ... +takes_value "port to be initialized")
+        (@arg pool_size: --("pool-size") +takes_value "memory pool size")
+        (@arg cache_size: --("cache-size") +takes_value "per core cache size")
+        (@arg dpdk_args: --("dpdk-args") ... +takes_value "custom DPDK arguments")
+        (@arg duration: -d --duration +takes_value "test duration")
+    )
+    .get_matches();
+}
+
+#[derive(Clone, Debug)]
+struct CommandLine();
+
+impl<'a> Source for CommandLine {
+    fn clone_into_box(&self) -> Box<Source + Send + Sync> {
```

Comment:
I will add a comment. basically I wanted to implement `Source` for `ArgMatches` so I can integrate `clap` with `config-rs`, but because I don't own the trait nor the struct, I had to go through this fake intermediary `CommandLine` to do that.